metrics/repl.lag/lag.go

```diff
@@ -198,3 +265,58 @@ func (c *Lag) Collect(ctx context.Context, levelName string) ([]blip.MetricValue
 	}
 	return []blip.MetricValue{m}, nil
 }
+
+func (c *Lag) collectLagFromPFS(ctx context.Context, levelName string) ([]blip.MetricValue, error) {
+	// if isReplCheck is supplied, check if it's a replica
+	defaultLag := func() ([]blip.MetricValue, error) {
+		if c.dropNotAReplica[levelName] {
+			return nil, nil
+		} else {
+			// send -1 for lag
+			m := blip.MetricValue{
+				Name:  "current",
+				Type:  blip.GAUGE,
+				Value: float64(-1),
+			}
+			return []blip.MetricValue{m}, nil
+		}
+	}
+	isRepl := 1
+	if c.replCheck != "" {
```

Comment:
I am not sure if we can skip this and just depend on the PFS query to determine if it's a replica.

Reply:
I don't know; try it out. Iirc this is mostly due to Aurora readers kind of looking like replicas but not.

Reply:
You are right, you can't tell if it's a reader or the cross-region writer of the replica from the output of the replica lag query from PF. So I will keep it around.